plugins/ocp/ocp-smart-extended-log.h

```diff
@@ -6,50 +6,100 @@
-	SCAO_MAVF	= 103,	/* Major Version Field */
-	SCAO_PCEC	= 104,	/* PCIe correctable error count */
-	SCAO_ICS	= 112,	/* Incomplete shutdowns */
-	SCAO_PFB	= 120,	/* Percent free blocks */
-	SCAO_CPH	= 128,	/* Capacitor health */
-	SCAO_NBEV	= 130,  /* NVMe Base Errata Version */
-	SCAO_NCSEV	= 131,  /* NVMe Command Set Errata Version */
-	SCAO_UIO	= 136,	/* Unaligned I/O */
-	SCAO_SVN	= 144,	/* Security Version Number */
-	SCAO_NUSE	= 152,	/* NUSE - Namespace utilization */
-	SCAO_PSC	= 160,	/* PLP start count */
-	SCAO_EEST	= 176,	/* Endurance estimate */
-	SCAO_PLRC	= 192,	/* PCIe Link Retraining Count */
-	SCAO_PSCC	= 200,	/* Power State Change Count */
-	SCAO_LPFR	= 208,	/* Lowest Permitted Firmware Revision */
-	SCAO_LPV	= 494,	/* Log page version */
-	SCAO_LPG	= 496,	/* Log page GUID */
+/**
+ * struct ocp_smart_extended_log -	SMART / Health Information Extended
+ * @physical_media_units_written:	Physical Media Units Written
+ * @physical_media_units_read:		Physical Media Units Read
+ * @bad_user_nand_blocks_raw:		Bad User NAND Blocks raw
+ * @bad_user_nand_blocks_normalized:	Bad User NAND Blocks normalized
+ * @bad_system_nand_blocks_raw:		Bad System NAND Blocks raw
+ * @bad_system_nand_blocks_normalized:	Bad System NAND Blocks normalized
+ * @xor_recovery_count:			XOR Recovery Count
+ * @uncorrectable_read_err_count:	Uncorrectable Read Error Count
+ * @soft_ecc_err_count:			Soft ECC Error Count
+ * @end_to_end_detected_err:		End to End detected errors
+ * @end_to_end_corrected_err:		End to End corrected errors
+ * @system_data_used_percent:		System data percent used
+ * @refresh_counts:			Refresh Counts
+ * @user_data_erase_count_max:		Max User data erase counts
+ * @user_data_erase_count_min:		Min User data erase counts
+ * @thermal_throttling_event_count:	Number of Thermal throttling events
+ * @dssd_errata_version:		DSSD Errata Version
+ * @dssd_point_version:			DSSD Point Version
+ * @dssd_minor_version:			DSSD Minor Version
+ * @dssd_major_version:			DSSD Major Version
+ * @pcie_correctable_err_count:		PCIe Correctable Error Count
+ * @incomplete_shoutdowns:		Incomplete Shutdowns
+ * @rsvd116:				Reserved
+ * @percent_free_blocks:		Percent free blocks
+ * @rsvd121:				Reserved
+ * @capacitor_health:			Capacitor health
+ * @nvme_base_errata_version:		NVM Express Base Errata Version
+ * @nvme_cmdset_errata_version:		NVMe Command Set Errata Version
+ * @rsvd132:				Reserved
+ * @unaligned_io:			Unaligned I/O
+ * @security_version:			Security Version Number
+ * @total_nuse:				Total NUSE - Namespace utilization
+ * @plp_start_count:			PLP start count
+ * @endurance_estimate:			Endurance Estimate
+ * @pcie_link_retaining_count:		PCIe Link Retraining Count
+ * @power_state_change_count:		Power State Change Count
+ * @lowest_permitted_fw_rev:		Lowest Permitted Firmware Revision -------------
+ * @rsvd216:				Reserved
+ * @log_page_version:			Log page version
+ * @log_page_guid:			Log page GUID
+ */
+struct __packed ocp_smart_extended_log {
```

Comment:
Thanks for the review.
I found out, after I read your review, there are few type cast I forgot to remove.
removed and uptaed.

since we need the __packed as below,  I didn't remove type cast from inside of le64_to_cpu
packed = 512
remove packed = 520

Reply:
Is the size increase because of the outer alignment? Could you post the output of pahole? Or did the spec screw this up? The `__packed` attribute should really be avoided if possible. It hides a lot of bugs.

Reply:
Hi @igaw 

Steven is on his honeymoon, so I fix it for him.
I removed the __packed keyword and aligned the struct size to be the same.
After #2577 merge, I will modify it if there is anything to modify.
Thanks.

Reply:
Nice I hope he doesn't get bothered by us here and can enjoy it! :)

Cool, so the __packed is not needed, this is what it what I hoped for. This will allow the compiler to report unaligned access etc. 

I'll go ahead with #2577 and wait for an update here. No hurry though, so if you don't have time on your hands don't worry.